aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1673,24 +1716,41 @@ Tensor sparse_compressed_to_sparse_bsc(const Tensor& self, IntArrayRef blocksize
                 "[blocksize=", blocksize,"] is not implemented.");
     return sparse_compressed_clone(self, blocksize, "to_sparse_bsc");
   }
-  AT_ERROR(
-      "sparse_compressed_to_sparse_bsc expected SparseBsr or SparseBsc layout but got ",
-      self.layout());
-  return self;
-}
-
-Tensor sparse_compressed_to_sparse_csc(const Tensor& self, c10::optional<int64_t> dense_dim_opt) {
-  if (dense_dim_opt.has_value()) {
-    AT_ERROR("sparse_compressed_to_sparse_csc conversion does not support specifying number of dense dimensions");
-  }
-  if (self.layout() == kSparseCsr) {
-    return sparse_compressed_to_flipped(self, c10::nullopt, "to_sparse_csc");
-  }
   if (self.layout() == kSparseCsc) {
-    return sparse_compressed_clone(self, c10::nullopt, "to_sparse_csc");
+    TORCH_CHECK(self.dim() == 2 + self.dense_dim(),
+                "to_sparse_bsc: conversion from Csc to Bsc for batched inputs is not implemented.");
+
+    Tensor self_values = self.values();
+    Tensor self_ccol_indices = self.ccol_indices();
+    Tensor self_row_indices = self.row_indices();
+
+    Tensor cpu_result = _compressed_to_block_compressed_cpu<false>(
+        _sparse_csc_tensor_unsafe(
+            self_ccol_indices.cpu(),
+            self_row_indices.cpu(),
+            self_values.cpu(),
+            self.sizes(),
+            self_values.scalar_type(),
+            self.layout(),
+            at::kCPU),
+        blocksize);
+
+    Tensor result_values = cpu_result.values().to(self_values.options());
+    Tensor result_ccol_indices =
+        cpu_result.ccol_indices().to(self_ccol_indices.options());
+    Tensor result_row_indices =
+        cpu_result.row_indices().to(self_row_indices.options());
+    return at::native::_sparse_bsc_tensor_unsafe(
+        result_ccol_indices,
+        result_row_indices,
+        result_values,
+        self.sizes(),
+        result_values.scalar_type(),
+        c10::kSparseBsc,
+        result_values.device());
```

Comment:
Same question as above: could we reduce this to:
```suggestion
    return _compressed_to_block_compressed_cpu<false>(self.cpu(), blocksize).to(self.device());
```

Reply:
Changed.  Thanks for all the comments!